packages/@react-native-windows/codegen/src/generators/GenerateTypeScript.ts

```diff
@@ -0,0 +1,229 @@
+        return `Array`;
+      }
+    case 'GenericObjectTypeAnnotation':
+      return 'object';
+    case 'ObjectTypeAnnotation':
+      return `{${type.properties
+        .map((prop: ObjectProp) => {
+          return `${prop.name}${prop.optional ? '?' : ''}: ${translateType(
+            prop.typeAnnotation,
+          )}`;
+        })
+        .join(', ')}}`;
+    case 'ReservedTypeAnnotation': {
+      // avoid: Property 'name' does not exist on type 'never'
+      const name = type.name;
+      // (#6597)
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (name !== 'RootTag')
+        throw new Error(
+          `Unknown reserved function: ${name} in translateReturnType`,
+        );
+      return 'RootTag';
+    }
+    case 'TypeAliasTypeAnnotation':
+      return type.name;
+    case 'NullableTypeAnnotation':
+      return `(${translateType(type.typeAnnotation)} | null | undefined)`;
+    case 'VoidTypeAnnotation':
+      return `void`;
+    case 'PromiseTypeAnnotation':
+      return `Promise`;
+    case `FunctionTypeAnnotation`:
+      return `((${type.params
+        .map((param: FunctionParam) => {
+          return `${param.name}${param.optional ? '?' : ''}: ${translateType(
+            param.typeAnnotation,
+          )}`;
+        })
+        .join(', ')}) => ${translateType(type.returnTypeAnnotation)})`;
+    default:
+      throw new Error(`Unhandled type in translateReturnType: ${returnType}`);
+  }
+}
+
+function translateAlias(
+  name: string,
+  type: NativeModuleObjectTypeAnnotation,
+): string {
+  return `interface ${name} {
+${type.properties
+  .map((prop: ObjectProp) => {
+    return `  ${prop.name}${prop.optional ? '?' : ''}: ${translateType(
+      prop.typeAnnotation,
+    )};`;
+  })
+  .join('\n')}
+}`;
+}
+
+function tryGetConstantType(
+  nativeModule: NativeModuleSchema,
+): NativeModuleObjectTypeAnnotation | undefined {
+  const candidates = nativeModule.spec.properties.filter(
+    (prop) => prop.name === 'getConstants',
+  );
+  if (candidates.length === 0) {
+    return undefined;
+  }
+
+  const getConstant = candidates[0];
+  const funcType =
+    getConstant.typeAnnotation.type === 'NullableTypeAnnotation'
+      ? getConstant.typeAnnotation.typeAnnotation
+      : getConstant.typeAnnotation;
+  if (
+    funcType.params.length > 0 ||
+    funcType.returnTypeAnnotation.type !== 'ObjectTypeAnnotation'
+  ) {
+    return undefined;
+  }
+
+  const constantType = funcType.returnTypeAnnotation;
+  if (constantType.properties.length === 0) {
+    return undefined;
+  }
+
+  return constantType;
+}
+
+function translateMethod(func: FunctionDecl): string {
+  const funcType =
+    func.typeAnnotation.type === 'NullableTypeAnnotation'
+      ? func.typeAnnotation.typeAnnotation
+      : func.typeAnnotation;
+
+  return `
+  ${func.name}(${funcType.params
+    .map((param: FunctionParam) => {
+      return `${param.name}${param.optional ? '?' : ''}: ${translateType(
```

Comment:
nit consider naming this, e.g.
```
      return `${param.name}${GetOptional(param)}: ${translateType(
```
you can reuse this GetOptional (or whatever you call it), below in 175 and several other places

Reply:
fixed